src/meta/src/rpc/server.rs

```diff
@@ -143,6 +141,19 @@ pub async fn rpc_serve_with_store<S: MetaStore>(
             let (leader_info, is_leader) = note_status_leader_rx.borrow().clone();
             let leader_addr = leader_info_to_host_addr(leader_info);
 
+            // Implementation of naive fencing mechanism:
+            // leader nodes should panic if they loose their leader position
+            // Current implementation panics, if leader looses lease.
+            // Alternative implementation that panics only if leader is unable to re-acquire lease:
+            // if was_leader && !is_leader {
+            if was_leader {
+                panic!(
+                    "This node lost its leadership. New host address is {}:{}. Killing node",
+                    leader_addr.host, leader_addr.port
```

Comment:
```suggestion
                    "This node lost its leadership. Last recorde leader at serving at {}:{}. Killing node",
                    leader_addr.host, leader_addr.port
```

I am not sure if we want to print this information. The address of the new leader may be incorrect. Picture the following scenario: 

- Network partition
- Leader can no longer access the meta store 
- Lease goes out of date 
- New leader is being elected 
- Old leader knows it is no longer leader, but it cannot get info on new leader
- This will read `"This node lost its leadership. New host address is old_leader_addr:port. Killing node"`



Maybe it is better to write: `Last recorded active leader at address:port`?

Reply:
I will remove this info. It is unreliable, because of the issues named above.